colour/io/uprtek_sekonic.py

```diff
@@ -0,0 +1,233 @@
+    def metadata(self):
+        """
+        Getter and setter property for the metadata.
+
+        Parameters
+        ----------
+        value : dict
+            Value to set the metadata with.
+
+        Returns
+        -------
+        unicode
+            Metadata.
+        """
+
+        return self._metadata
+
+    def read(self):
+        """
+        Reads and parses the spectral data from a given *UPRTek* CSV file.
+
+        Returns
+        -------
+        bool
+            Definition success.
+
+        Examples
+        --------
+        >>> from os.path import dirname, join
+        >>> directory = dirname(__file__)
+        >>> sd = SpectralDistribution_UPRTek(join(directory, 'upr.xls.txt'))
+        >>> print(sd.read())
+        [[  3.80000000e+02   3.02670000e-02]
+        [  3.81000000e+02   3.02670000e-02]
+        [  3.82000000e+02   3.02670000e-02]
+        [  3.83000000e+02   2.98220000e-02]
+        ...
+        ...
+        [  7.78000000e+02   4.11766000e-01]
+        [  7.79000000e+02   4.11766000e-01]
+        [  7.80000000e+02   4.11766000e-01]]
+        >>> sd['400']
+        0.019325
+        """
+
+        def as_array(a):
+            return [float(e) for e in a]
+
+        spectral_sections = defaultdict(list)
+        with open(self.path, encoding="utf-8") as csv_file:
+            content = csv.reader(csv_file, delimiter=self._delimiter)
+
+            spectral_section = 0
+            for row in content:
+                if not "".join(row).strip():
+                    continue
+
+                key, value = row[0], row[1:]
+                value = value[0] if len(value) == 1 else value
+
+                search = re.search("(\\d{3})\\[?nm\\]?", key)
+                if search:
+                    wavelength = search.group(1)
+
+                    if wavelength == self._spectra_section:
+                        spectral_section += 1
+
+                    spectral_sections[spectral_section].append(
+                        [wavelength, value])
+                else:
+                    for method in (int, float, as_array):
+                        try:
+                            self._metadata[key] = method(value)
+                            break
+                        except Exception:
+                            self._metadata[key] = value
+
+        self.name = os.path.splitext(os.path.basename(self.path))[0]
+        spectral_data = as_float_array(spectral_sections[sorted(
+            spectral_sections.keys())[-1]])
+
+        self.wavelengths = spectral_data[..., 0]
+        self.values = spectral_data[..., 1]
+
+        self.header.comments = json.dumps(self._metadata)
+        self.header.report_date = self._metadata["Time"][:10]
+        self.header.measurement_equipment = self._metadata["Model Name"]
+        self.header.manufacturer = "UPRTek"
+        self.spectral_quantity = 'Irradiance'
+
+        return self
+
+
+class SpectralDistribution_Sekonic(SpectralDistribution_UPRTek):
```

Comment:
Will need docstrings, doctests and  unit tests here.

Reply:
1. I had a doubt. For the UPRTek parser, since we are inheriting the `write` method, and not modifying it in the child class in any way, do I still need to write unit tests for it?
2. The same for the Sekonic parser. Since we are not introducing any new methods in it, do I still need to write unit tests for it? I left the docstring, doctests and unit tests for sekonic because I had this doubt. 
3. Also, since there are two different spectral data that we get in a sekonic csv file, how do we access the other one?

Reply:
1. All good with the write method, we can do without!
2. Same for the write method, however, we need basic testing to be sure that it works.
3. All good to lose one.